src/freenet/support/SortedIntSet.java

```diff
@@ -93,44 +99,22 @@ public synchronized boolean contains(int num) {
 	 *            the item to be removed
 	 */
 	public synchronized boolean remove(int item) {
-		boolean ret = false;
 		int x = binarySearch(item);
 		if(x >= 0) {
-			if(x < length-1)
-				System.arraycopy(data, x+1, data, x, length-x-1);
-			data[--length] = Integer.MAX_VALUE;
-			ret = true;
+			return data.remove(x) != null;
```

Comment:
How did you arrive at that `!= null` expression? I can’t really spot anything similar in the original code…